docs/user_guide/datasets.rst

```diff
@@ -0,0 +1,333 @@
+Datasets
+==========
+
+Revisiting the Boston Housing Dataset
+-------
+
+Introduction
+^^^^^^^^^^^^^^^^^
+
+The Boston Housing dataset is one of three datasets currently callable in 
```

Comment:
nit: since we may add more datasets, this text is almost certain to get outdated without us noticing.
```suggestion
The Boston Housing dataset is one of the datasets currently callable in 
```